near-sdk/src/json_types/public_key.rs

```diff
@@ -39,24 +57,23 @@ impl std::str::FromStr for CurveType {
 ///
 /// # Example
 /// ```
-/// use near_sdk::json_types::Base58PublicKey;
+/// use near_sdk::json_types::PublicKey;
 ///
 /// // Compressed ed25519 key
-/// let ed: Base58PublicKey = "ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp".parse()
+/// let ed: PublicKey = "ed25519:6E8sCci9badyRkXb3JoRpBj5p8C6Tw41ELDZoiihKEtp".parse()
 ///             .unwrap();
 ///
 /// // Uncompressed secp256k1 key
-/// let secp256k1: Base58PublicKey  = "secp256k1:qMoRgcoXai4mBPsdbHi1wfyxF9TdbPCF4qSDQTRP3TfescSRoUdSx6nmeQoN3aiwGzwMyGXAb1gUjBTv5AY8DXj"
+/// let secp256k1: PublicKey = "secp256k1:qMoRgcoXai4mBPsdbHi1wfyxF9TdbPCF4qSDQTRP3TfescSRoUdSx6nmeQoN3aiwGzwMyGXAb1gUjBTv5AY8DXj"
 ///             .parse()
 ///             .unwrap();
 /// ```
-#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq)]
-pub struct Base58PublicKey {
-    curve: CurveType,
+#[derive(Debug, Clone, PartialEq, PartialOrd, Ord, Eq, BorshDeserialize, BorshSerialize)]
+pub struct PublicKey {
```

Comment:
This should not be exported as `near_sdk::json_types::PublicKey` and should not live in this module at all. Just need the `Base58PublicKey` alias to live here as deprecated until we remove it. The goal of this is to move away from the `json_types` module

Reply:
I see, I'll just move to the whole entire thing to `types/public_key.rs`. I was thinking of re-exporting CurveType from there too, and adding a deprecated attribute, but deprecated doesn't work on exports just yet